test/lit/passes/denan-simd.wast

```diff
@@ -0,0 +1,131 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; RUN: foreach %s %t wasm-opt --denan -all -S -o - | filecheck %s
+
+(module
+  ;; CHECK:      (type $0 (func (param v128) (result v128)))
+
+  ;; CHECK:      (type $1 (func (param f32) (result f32)))
+
+  ;; CHECK:      (type $2 (func (param f64) (result f64)))
+
+  ;; CHECK:      (func $foo128 (type $0) (param $x v128) (result v128)
+  ;; CHECK-NEXT:  (local.set $x
+  ;; CHECK-NEXT:   (call $deNan128
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (call $deNan128
+  ;; CHECK-NEXT:    (v128.const i32x4 0x00000001 0x00000002 0x00000003 0x00000004)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (v128.const i32x4 0x00000000 0x00000000 0x00000000 0x00000000)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (v128.const i32x4 0x00000000 0x00000000 0x00000000 0x00000000)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $deNan128
+  ;; CHECK-NEXT:   (call $foo128
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $foo128 (param $x v128) (result v128)
+    ;; The incoming param will be de-naned.
+
+    ;; This is not a NaN. (We do still emit a call for it atm, FIXME)
```

Comment:
Why does this happen?

Reply:
No good reason - that's what the code did for f32 and f64 before this PR. I can fix it later.